nautobot/core/views/generic.py

```diff
@@ -1002,11 +1007,12 @@ def post(self, request, **kwargs):
                                     setattr(obj, name, form.cleaned_data[name])
 
                             # Update custom fields
-                            for name in custom_fields:
-                                if name in form.nullable_fields and name in nullified_fields:
-                                    obj.cf[name] = None
-                                elif form.cleaned_data.get(name) not in (None, ""):
-                                    obj.cf[name] = form.cleaned_data[name]
+                            for field_name in form_custom_fields:
+                                # 2.0 TODO: #824 when we use slug in obj.cf we can just do obj.cf[field_name[3:]]
+                                if field_name in form.nullable_fields and field_name in nullified_fields:
+                                    obj.cf[form_cf_to_key[field_name]] = None
+                                elif form.cleaned_data.get(field_name) not in (None, ""):
+                                    obj.cf[form_cf_to_key[field_name]] = form.cleaned_data[field_name]
```

Comment:
```suggestion
                                if field_name in form.nullable_fields and field_name in nullified_fields:
                                    val = None
                                elif form.cleaned_data.get(field_name) not in (None, ""):
                                    val = form.cleaned_data[field_name]
                                obj.cf[form_cf_to_key[field_name]] = val
```

Reply:
But it's one line longer! :-) Will change.

Reply:
Actually, no - in the current logic we only update `obj.cf` if one of the two `if/elif` conditions is met. Changing this as suggested would cause `obj.cf` to be updated unconditionally.